mounts.go

```diff
@@ -151,8 +151,7 @@ func (cm *CacheMap) LoadFromFile(stateFile string, cacheDir string) {
 func (feature *MountsFeature) Initialise() error {
 	fileCaches.LoadFromFile("file-caches.json", config.CachesDir)
 	directoryCaches.LoadFromFile("directory-caches.json", config.DownloadsDir)
-	pc = tcpurgecache.New(nil)
-	pc.BaseURL = config.PurgeCacheBaseURL
+	pc = tcpurgecache.New(&tcclient.Credentials{RootURL: config.TaskclusterRootURL})
```

Comment:
This is a good example of why I suggest that the client's `New` fail if `RootUrl` isn't provided -- unauthenticated requests like this would otherwise continue to work, and in this case at least not even cause any visible failures (workers in any cluster would just consult the https://taskcluster.net purge-cache instance -- until the day we shut that off).

The automatic use of env vars by the client makes the risk a little worse, too -- I expect `tcpurgecache.New(nil)` would work as long as TASKCLUSTER_ROOT_URL was set, which it probably is during testing but probably is not in production.  In the JS tests, this caused some unhappiness, and we now [unset the environment variables](https://github.com/taskcluster/taskcluster-lib-testing/commit/05023065ea1435f95c789ab6d40a4b6079ca1f40) after reading them into the config.  So that's something to watch out for.